### ISSUE TITLE
Question about wibox.widget.textbox:set_font(font) and other similar methods

### ISSUE THREAD

**Issue by Sorky** [awesomeWM/awesome#2476]
@psychon

What do you (and others) think of...

    function textbox:set_font(font)
        self._private.layout:set_font_description(beautiful.get_font(font))
        self:emit_signal("widget::redraw_needed")
        self:emit_signal("widget::layout_changed")
        **return self**
    end

To allow...

	local textbox = wibox.widget.textbox( "??" ):set_font( font )

PS: This method doesn't seem to appear in the doco

**Comment by Sorky** [comment]
I was able to do 
    local textbox = wibox.widget.textbox( "Hi" )
    textbox.font = font
and
    local textbox = wibox.widget.textbox( "Hi" )
    textbox:set_font( font )

Surely the method is additional in this case? Causes a refresh?

**Comment by Sorky** [comment]
Just like lua string methods

**Comment by Elv13** [comment]
> Just like lua string methods

No. Lua string return a new string, not a reference to the old string.
